src/app/lib/models/generic_collection.js

```diff
@@ -52,7 +52,8 @@
 
                         return;
                     }
-                    movie.provider = torrentProvider.name;
+
+                    movie.provider = torrentProvider;
```

Comment:
Should remain torrentProvider.name, torrentProvider is an object. Or if you want to change that (see comment below), there is more than just here